src/main/java/hudson/plugins/jabber/im/transport/JabberIMConnection.java

```diff
@@ -312,13 +311,25 @@ private boolean createConnection() throws XMPPException, SaslException,
 		
 		String serviceName = desc.getServiceName();
 		final XMPPTCPConnectionConfiguration.Builder cfg = XMPPTCPConnectionConfiguration.builder();
+		
+		if (pi.getProxyType() == ProxyType.NONE) {
+			cfg.setSocketFactory(SocketFactory.getDefault());
+		} else {
+			cfg.setProxyInfo(pi);
+		}
+		
 		if (serviceName == null) {
-			cfg.setHost(hostnameOverride).setPort(port).setProxyInfo(pi);
+			cfg
+			    .setHost(hostnameOverride)
+			    .setPort(port);
 		} else if (this.hostnameOverride == null) {
-		    // uses DNS lookup, to get the actual hostname for this service: 
-			cfg.setServiceName(serviceName).setProxyInfo(pi);
+		    // uses DNS lookup, to get the actual hostname for this service:			
+			cfg.setServiceName(serviceName);
 		} else {
-			cfg.setHost(hostnameOverride).setPort(port).setServiceName(serviceName).setProxyInfo(pi);
+			cfg
```

Comment:
But to me, it is a lot clearer what happens here (more so if you look at the resulting file rather than only the diff). But I appreciate that this is in the eye of the beholder